applications/IgaApplication/custom_conditions/coupling_nitsche_condition.cpp

```diff
@@ -0,0 +1,1440 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ricky Aristio
+//                   
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_conditions/coupling_nitsche_condition.h"
+#include "geometries/brep_curve_on_surface.h"
+
+namespace Kratos
+{
+    ///@name Initialize Functions
+    ///@{
+
+    void CouplingNitscheCondition::Initialize(const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY
+
+        InitializeMaterial();
+
+        KRATOS_CATCH("")
+    }
+
+    void CouplingNitscheCondition::InitializeMaterial()
+    {
+        KRATOS_TRY
+
+        const auto& r_geometry_master = GetGeometry().GetGeometryPart(0);
+        const auto& r_geometry_slave = GetGeometry().GetGeometryPart(1);
+        const Properties& r_properties = GetProperties();
+        const auto& r_N_master = r_geometry_master.ShapeFunctionsValues();
+        const auto& r_N_slave = r_geometry_slave.ShapeFunctionsValues();
+
+        const SizeType r_number_of_integration_points_master = r_geometry_master.IntegrationPointsNumber();
+        const SizeType r_number_of_integration_points_slave = r_geometry_slave.IntegrationPointsNumber();
+
+        //Constitutive Law initialisation
+        if (mConstitutiveLawVector_master.size() != r_number_of_integration_points_master)
```

Comment:
I would recommend just using the constitutive law vector and afterwards only checking the specific mast or slave one. In that way you need a little bit less storage.

Reply:
There may be a case where the constitutive law vectors may be different for both master and slave. Well, I open to discuss this in detail.

Reply:
> There may be a case where the constitutive law vectors may be different for both master and slave. Well, I open to discuss this in detail.

Sorry for the misunderstanding. I've changed how the constitutive law vector is assigned in the recent commit.

Reply:
No worries, thanks a lot for the change!